src/Nest/Search/Search/Sort/SortBase.cs

```diff
@@ -4,47 +4,103 @@
 
 namespace Nest
 {
+	/// <summary>
+	/// Allows to add one or more sort on specific fields.
+	/// </summary>
 	[JsonObject(MemberSerialization = MemberSerialization.OptIn)]
 	[ContractJsonConverter(typeof(SortJsonConverter))]
 	public interface ISort
 	{
+		/// <summary>
+		/// The field to sort on
+		/// </summary>
 		Field SortKey { get; }
 
-		[JsonProperty("missing")]
+		/// <summary>
+		/// Specifies how docs which are missing the field should be treated
+		/// </summary>
+		[JsonIgnore]
+		[Obsolete("Use MissingValue")]
 		string Missing { get; set; }
 
+		/// <summary>
+		/// Specifies how docs which are missing the field should be treated
+		/// </summary>
+		[JsonProperty("missing")]
+		object MissingValue { get; set; }
+
+		/// <summary>
+		/// The sort order
+		/// </summary>
 		[JsonProperty("order")]
 		SortOrder? Order { get; set; }
 
+		/// <summary>
+		/// Elasticsearch supports sorting by array or multi-valued fields. <see cref="Mode"/>
+		/// controls what array value is picked for sorting the document it belongs to.
+		/// </summary>
 		[JsonProperty("mode")]
 		SortMode? Mode { get; set; }
 
+		/// <summary>
+		/// A filter that the inner objects inside the nested path should match with in order for its field values
+		/// to be taken into account by sorting.
+		/// Common case is to repeat the query / filter inside the nested filter or query.
+		/// </summary>
 		[JsonProperty("nested_filter")]
 		QueryContainer NestedFilter { get; set; }
 
+		/// <summary>
+		/// Defines on which nested object to sort. The actual sort field must be a direct field inside
+		/// this nested object. When sorting by nested field, this field is mandatory.
+		/// </summary>
 		[JsonProperty("nested_path")]
 		Field NestedPath { get; set; }
 	}
 
+	/// <inheritdoc/>
 	public abstract class SortBase : ISort
 	{
-		public string Missing { get; set; }
+		/// <inheritdoc/>
+		[Obsolete("Use MissingValue")]
+		public string Missing
+		{
+			get => MissingValue as string;
```

Comment:
`.ToString()` ?

Reply:
These shouldn't be `.ToString()` in my opinion because it means a non-string value set with `MissingValue` can be retrieved as a `string` using `Missing`.